### PR TITLE
v1.11.0 – CSS principles, structure, sass, naming and fozzie sections added/updated

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- Sass docs page added
- Fozzie docs page added

### Updated
- CSS Principles docs page now up-to-date
- CSS Naming Scheme docs page now up-to-date
- CSS Structure docs page now up-to-date
- Footer for documentation pages added to the site
- Updating navigation and homepage with more accurate/useful descriptions
- Changed `max-width` of content docs to be a more sensible line-length

### Fixed
- UI Components page fixed to include current UI Components
- `espree` added to resolutions to fix ESLint dependency issue

### Removed
- gift `resolutions` value for `fozzie-colour-palette` no longer needed
- Removed markup from base design elements page (as not needed)

### CODE REVIEW

File: docs/src/templates/includes/documentation-components/footer.hbs [hunk 1]
```diff
@@ -2,6 +2,10 @@
 		<!--  ==============  -->
 		<!--  === Footer ===  -->
 		<!--  ==============  -->
-		<footer class="footer" role="contentinfo">
-
-		</footer>
\ No newline at end of file
+		<footer class="sg-footer" role="contentinfo">
+            <div class="l-container text-center">
+                <p>Fozzie and it's associated documentation was created with ❤️ by the UI team at Just Eat.</p>
```
Comment (xander-marjoram) [inline docs/src/templates/includes/documentation-components/footer.hbs]: "its", rather than "it's"

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,32 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+
+v1.11.0
+------------------------------
+*August 15, 2018*
```
Comment (xander-marjoram) [inline CHANGELOG.md]: 📆

File: docs/src/templates/pages/documentation/css/css-naming.hbs [hunk 3]
```diff
@@ -57,51 +57,81 @@ Using a naming scheme helps to document the intent and relationships from our cl
...
 
 
-```
+<a name="modifiers"></a><h2 class="sg-sectionHeading">Modifier Classes</h2>
 
-## State Classes
+Modifier classes are used to show when an object or component class is being modified slightly.  Think of them as helping to define one or more extensions to your component.
 
-We use the `is-` or `has-` prefixes to indicate state.
+A typical example is when defining buttons.  Usually all buttons share common styles, that are defined using the `o-btn` class.  However, buttons usually have various design treaments – some may have a different background colour and font-size, others may have rounded corners.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/css-naming.hbs]: "treaments" -> "treatments"

File: docs/src/templates/pages/documentation/css/fozzie.hbs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+<a name="modules"></a><h2 class="sg-sectionHeading">Modules</h2>
+
+To see the current list of public Fozzie modules, take a look on [our public Github account](https://github.com/search?q=topic%3Afozzie+org%3Ajusteat&type=Repositories).
+
+
+<a name="import"></a><h2 class="sg-sectionHeading">Importing Fozzie Modules</h2>
+
+To help with importing fozzie SCSS styles, we use a tool called [**Eyeglass**](https://github.com/sass-eyeglass/eyeglass).
+
+Essentially, Eyeglass let's us import modules in our Sass by referencing the name of the module, rather than having to provide absolute paths into the `node_modules` folder.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/fozzie.hbs]: "let's" -> "lets"

File: docs/src/templates/pages/documentation/css/index.hbs [hunk 5]
```diff
@@ -1,141 +1,59 @@
 ---
 title: CSS Principles
 section-title: CSS Documentation
-description: Our commitment to creating an accessible platform
+description: The principles around how we use CSS at Just Eat
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/index.hbs]: This seems a bit wordy, how about just "How we use CSS at Just Eat"

File: docs/src/templates/pages/documentation/css/index.hbs [hunk 6]
```diff
@@ -1,141 +1,59 @@
...
-
-We encourage this way of thinking at JUST EAT and our Sass is [structured to reflect this](#structure), with base styles defined globally and individual components and modules that extend upon these styles.
+There are many CSS methodologies that support this way of thinking, such as [SMACSS](https://smacss.com/), [Atomic Design](http://bradfrost.com/blog/post/atomic-web-design/)) and [ITCSS](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/).  These all share a lot of common ground whn it comes to constructing good CSS; one of main themes being that you should always think about how you can write small reusable components that can be easily extended later.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/index.hbs]: "whn" -> "when"
And a comma between small and reusable while you're at it, please!

File: docs/src/templates/pages/documentation/css/sass.hbs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+description: How we use Sass on projects at Just Eat
+docs: true
+
+navgroup: documentation
+navsub: docs-css
+navactive: css-sass
+---
+
+
+We use [Sass](http://sass-lang.com/) to compliment how we write our CSS, using it’s features to make our front-end code more maintainable and to give it better structure.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/sass.hbs]: "compliment" -> "complement"
"it's" -> "its"

File: docs/src/templates/pages/documentation/css/sass.hbs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+When using these mixins, the easiest way to write your media queries is to specify the breakpoint name; the @include-media library will then convert this into a media query with a value in ems to provide a more consistent experience for users that use text zoom.
+
+
+### Media query example
+
+```scss
+/**
+ * Using @include-media you can specify breakpoints
+ * using the map key's defined in Fozzie's [breakpoint map](https://github.com/justeat/fozzie/blob/8a1094ae5a467e9e2afd86a4aab62f9ea65b1bee/src/scss/settings/_variables.scss#L65-L72)
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/sass.hbs]: "key's" -> "keys"

File: docs/src/templates/pages/documentation/css/sass.hbs [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+Colour variables are managed in the [fozzie-colour-palette](https://github.com/justeat/fozzie-colour-palette) module.
+
+This helps us version our colour palette independently to any of our other code and allows us to roll out brand colour changes quickly across projects hooked into this module.
+
+---
+
+### Fozzie Variables
+
+The other main framwork variables are stored directly in the [base fozzie module](https://github.com/justeat/fozzie), in the [_variables.scss file](https://github.com/justeat/fozzie/blob/master/src/scss/settings/_variables.scss).
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/sass.hbs]: "framwork" -> "framework"

File: docs/src/templates/pages/documentation/css/structure.hbs [hunk 10]
```diff
@@ -6,112 +6,134 @@ docs: true
...
 
-The [modules](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials/modules) folder is where components that would be specific to the project are located, such as [navigation](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials/modules/_nav.scss) and [social-login](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials/modules/_social-login.scss).
+<h3 class="sg-sectionHeading">Components</h3>
 
-**Please [browse through](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials/) the included partials to see what is already defined before adding your own**.
+The [components folder](https://github.com/justeat/fozzie/tree/master/src/scss/components) should be used to define custom component styles specific to that module.  In the fozzie base module, this includes components such as alerts, breadcrumbs and rating components, among others.
 
-<hr>
+Component filenames should closely relate to their component name. So a component of `c-rating` should have it's styles in a file called `_ratings.scss`
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/structure.hbs]: "it's" -> "its"

File: docs/src/templates/pages/documentation/css/structure.hbs [hunk 11]
```diff
@@ -6,112 +6,134 @@ docs: true
...
-The [functions](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/functions/) folder contains various Sass functions that are used in the base framework that you might find useful.
+The most common example of this is CSS variables, such as font-sizes, grid settings and breakpoint definitions.
 
-<hr>
 
-### Important files
+<h3 class="sg-sectionHeading">Tools</h3>
 
-It’s important to become familiar with **all of these files** so that you can make full use of the framework.
+The [tools folder](https://github.com/justeat/fozzie/tree/master/src/scss/tools) is where globally used mixins and functions are located.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/structure.hbs]: "globally used" -> "globally-used" or "global"?

File: docs/src/templates/pages/documentation/css/structure.hbs [hunk 12]
```diff
@@ -6,112 +6,134 @@ docs: true
...
 
-**Try not to fill this up with all your styles though.**  Your Sass should be written in a modular way, and so the majority of your Sass should be organised within the `partials` or `views` folders.
+<a name="key-files"></a><h2 class="sg-sectionHeading">Key files</h2>
+
+Within the base fozzie module, there are a few key files that if you want to contribute back into fozzie, it's worth understanding.
+
+
+<h3 class="sg-sectionHeading"><a href="https://github.com/justeat/fozzie/blob/master/src/scss/fozzie.scss"><strong>fozzie.scss</strong></a></h3>
+
+This is the base Sass file of the fozzie module.  When using [Eyeglass]({{ ../baseUrl }}/documentation/css/fozzie#import), this is the route of the module that gets pulled into your project.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/css/structure.hbs]: "route" -> "root"?

File: docs/src/templates/includes/documentation-components/footer.hbs [hunk 13]
```diff
@@ -2,6 +2,10 @@
 		<!--  ==============  -->
 		<!--  === Footer ===  -->
 		<!--  ==============  -->
-		<footer class="footer" role="contentinfo">
-
-		</footer>
\ No newline at end of file
+		<footer class="sg-footer" role="contentinfo">
+            <div class="l-container text-center">
```
Comment (oanuna) [inline docs/src/templates/includes/documentation-components/footer.hbs]: Tabulation?

File: docs/src/templates/includes/documentation-components/nav.hbs [hunk 14]
```diff
@@ -6,9 +6,10 @@
     <input type="checkbox" class="c-nav-trigger" name="menuVisible" id="menu-visible" aria-hidden="true">
     <div class="c-nav-container">
     	<ul class="c-nav-list">
-    		<li class="c-nav-list-item{{#is navgroup 'styleguide'}} c-nav-list-item--active{{/is}}"><a href="{{ baseUrl }}/styleguide/design-elements/colours" class="c-nav-list-link">Styleguide &amp; Components</a></li>
-    		<li class="c-nav-list-item{{#is navgroup 'documentation'}} c-nav-list-item--active{{/is}}"><a href="{{ baseUrl }}/documentation/general/" class="c-nav-list-link">Documentation</a></li>
-    		<li class="c-nav-list-item{{#is navgroup 'perf'}} c-nav-list-item--active{{/is}}"><a href="{{ baseUrl }}/performance/" class="c-nav-list-link">Performance Monitoring</a></li>
+    		<li class="c-nav-list-item{{#is navgroup 'styleguide'}} c-nav-list-item--active{{/is}}"><a href="{{ baseUrl }}/styleguide/design-elements/colours" class="c-nav-list-link">Design Elements</a></li>
+            <li class="c-nav-list-item{{#is navgroup 'styleguide'}} c-nav-list-item--active{{/is}}"><a href="{{ baseUrl }}/styleguide/ui-components/" class="c-nav-list-link">UI Components</a></li>
```
Comment (oanuna) [inline docs/src/templates/includes/documentation-components/nav.hbs]: Tabulation?

File: docs/src/templates/includes/documentation-components/subnavs/subnav--sidebar.hbs [hunk 15]
```diff
@@ -160,9 +160,52 @@
...
+        {{#is navactive 'css-principles'}}
+        <ul class="sg-sideNav-subnav u-unstyled">
+            <li><a href="#philosophy">Philosophy</a></li>
+            <li><a href="#bestpractices">Best Practices</a></li>
+            <li><a href="#preprocessors">Preprocessors</a></li>
+            <li><a href="#progressive">Progressive Enhancement</a></li>
+        </ul>
+        {{/is}}
+    </li>
+    <li><a href="{{ baseUrl }}/documentation/css/sass" class="{{#is navactive 'sass'}} is-active{{/is}}">Sass</a>
```
Comment (oanuna) [inline docs/src/templates/includes/documentation-components/subnavs/subnav--sidebar.hbs]: should all anchor tags be on a new line?

File: package.json [hunk 16]
```diff
@@ -39,7 +39,7 @@
     "gulp": "^3.9.1"
   },
   "resolutions": {
-    "@justeat/gulp-build-fozzie/gulp-gh-pages/gift": "0.10.2"
+    "espree": "^3.5.4"
```
Comment (oanuna) [inline package.json]: we should fix the verison to `3.5.4`

File: docs/src/templates/includes/documentation-components/subnavs/subnav--sidebar.hbs (hunk 15 above)
Comment (ashleynolan) [reply docs/src/templates/includes/documentation-components/subnavs/subnav--sidebar.hbs]: It's not something we've discussed in terms of whether we do this or not.

I'm pretty easy either way – as if it's all on one line, looks tidier to me, but can see it from the other side too where people like new tags on new lines.

**Comment by ashleynolan** [comment]
@xander-marjoram @oanuna Have addressed review comments

File: docs/src/templates/pages/documentation/css/index.hbs (hunk 6 above)
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/index.hbs]: "whn it comes" -> "when it comes"

File: docs/src/templates/pages/documentation/css/index.hbs [hunk 17]
```diff
@@ -1,141 +1,59 @@
...
+We strongly believe in using progressive enhancement so that we can provide a service to as many users as we can.  Those using modern browsers will get the best possible experience, with those on [older browsers](/general/browser-support) still able to view and use our site.
 
-Our minimal framework includes a custom build of Modernizr. To update this with an extra feature test that isn’t already included in our build, ensure you copy the url in the current [modernizr.min.js](https://github.je-labs.com/I18N-Production-ROTW/ConsumerWeb/blob/master/src/ConsumerWeb/Assets/src/js/libs/modernizr.min.js) so that the current tests are preserved in the new build.
+As part of this effort, we strongly recommend using feature detection over browser or UI sniffing.
 
-For full information on using Modernizr, see [their documentation](http://modernizr.com) or see our [Javascript documentation]({{ baseUrl }}/js/#js-modernizr) to find out where it is included.
+Where needed, we use Modernizr to help us do this, and where possible [CSS feature queries via `@supports`](https://caniuse.com/#feat=css-featurequeries) (browser support allowing – ironically).
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/index.hbs]: Is this still true for the global microsites?

File: docs/src/templates/pages/documentation/css/sass.hbs (hunk 9 above)
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/sass.hbs]: "framwork" -> "framework"

File: docs/src/templates/pages/documentation/css/sass.hbs [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+
+Examples of the variables stored here include (but is not limited to):
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/sass.hbs]: "are *not* limited to"?

File: docs/src/templates/pages/documentation/css/structure.hbs [hunk 19]
```diff
@@ -6,112 +6,134 @@ docs: true
...
+<h3 class="sg-sectionHeading">Base</h3>
 
-<hr>
+The [base folder](https://github.com/justeat/fozzie/tree/master/src/scss/base) is where the styles that make up the foundation of the module are defined.
 
-### /partials
+For example, typographic styles can be defined here, as can base layout, reset and print styles.
 
-The [partials](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials) folder contains two folders; `/components` and `/modules`.
+n.b. We use [Normalise.css](https://github.com/justeat/fozzie/blob/master/src/scss/_dependencies.scss#L36) over a standard reset stylesheet, but styles that are need on top of this to 'reset' element styling can be added in a module's reset file here.
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/structure.hbs]: Use the canonical URL here in case the code changes to ensure the user is taken to the correct line.

File: docs/src/templates/pages/documentation/css/structure.hbs [hunk 20]
```diff
@@ -6,112 +6,134 @@ docs: true
...
+<h3 class="sg-sectionHeading">Base</h3>
 
-<hr>
+The [base folder](https://github.com/justeat/fozzie/tree/master/src/scss/base) is where the styles that make up the foundation of the module are defined.
 
-### /partials
+For example, typographic styles can be defined here, as can base layout, reset and print styles.
 
-The [partials](https://github.je-labs.com/ICP/app_consumerweb/tree/master/src/ConsumerWeb/Assets/src/scss/partials) folder contains two folders; `/components` and `/modules`.
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/structure.hbs]: "are need on top" -> "are *needed* on top"

File: docs/src/templates/pages/documentation/css/index.hbs (hunk 17 above)
Comment (ashleynolan) [reply docs/src/templates/pages/documentation/css/index.hbs]: I've left this in for now – we don't atm but until the GlobalWeb pages are built out I didn't know if this would be irrelevant or not, so thought it best to keep it in!

File: docs/src/templates/pages/documentation/css/sass.hbs (hunk 9 above)
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/css/sass.hbs]: Wrap _variables.scss in backticks.

File: docs/src/templates/pages/documentation/css/structure.hbs (hunk 20 above)
Comment (ashleynolan) [reply docs/src/templates/pages/documentation/css/structure.hbs]: Done

**Comment by ashleynolan** [comment]
@DamianMullins Have fixed the comments – let me know if that looks ok
